github/Branch.py

```diff
@@ -180,14 +181,25 @@ def edit_protection(
         assert is_optional(allow_deletions, bool), allow_deletions
 
         post_parameters: dict[str, Any] = {}
-        if is_defined(strict) or is_defined(contexts):
+        if is_defined(strict) or is_defined(contexts) or is_defined(checks):
             if is_undefined(strict):
                 strict = False
-            if is_undefined(contexts):
-                contexts = []
+            if is_undefined(contexts) and is_undefined(checks):
+                checks = []
+            elif is_defined(checks):
+                post_parameters["checks"] = []
+                for check in checks:
+                    if isinstance(check, tuple):
+                        context, app_id = check
+                        post_parameters["checks"].append({"context": context, "app_id": app_id})
+                    else:
+                        post_parameters["checks"].append({"context": check})
+            elif is_defined(contexts):
+                post_parameters["checks"] = [{"context": context} for context in contexts]
```

Comment:
I think you want to populate list `checks` here, as used below to set `post_parameters["required_status_checks"]`.